docs/network-flow-visibility.md

```diff
@@ -610,3 +613,75 @@ and other Theia features, please refer to the
 instead, which is actively maintained.
+
+## L7 Visibility
+
+In addition to Layer 4 network visibility, Antrea adds Layer 7 network flow visibility
+enabling users to collect and visualize application-layer flow data in Theia.
+
+### Prerequisites
+
+To achieve L7 (Layer 7) network flow visibility, the `L7Visibility` feature gate must be enabled.
+Due to the constraint of the application detection engine, TX checksum offloading must be disabled via the
+`disableTXChecksumOffload` option in antrea-agent.conf for the feature to work. An example configuration is as below:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: antrea-config
+  namespace: kube-system
+data:
+  antrea-agent.conf: |
+    disableTXChecksumOffload: true
+    featureGates:
+      L7Visibility: true
+```
+
+Alternatively, you can use the following helm installation command to configure the above options:
+
+```bash
+helm install antrea antrea/antrea --namespace kube-system --set featureGates.L7Visibility=true,disableTXChecksumOffload=true
+```
+
+### Usage
+
+To export layer 7 flows of a Pod or Namespace, user can annotate Pods or Namespaces
+with the annotation key `visibility.antrea.io/enable-l7` and set the value to indicate
+the traffic flow direction, which can be `ingress`, `egress` or `both`.
+
+For example, to enable L7 flow visibility in the ingress direction on
+Pod test-pod in the default Namespace, you can use:
+
+```bash
+kubectl annotate pod test-pod visibility.antrea.io/enable-l7=ingress
+```
+
+Based on the annotation, Flow Exporter will export the L7 flow data to
+the Flow Aggregator using the fields `l7ProtocolName` and `httpVals`.
+
+* `l7ProtocolName` field is used to store application layer protocol name,
+field will be empty if application layer flow is not exported.
+* `httpVals` stores a serialized JSON object with every HTTP flow mapped to the TxID.
+Mapping the HTTP flows to the TxID helps store all HTTP flows for a connection
+in a single row. Layer 7 flows follows the same path as other Layer 4 flows
+as elaborated above.
+
+An example of `httpVals` is :
+
+`"{\"0\":{\"hostname\":\"10.10.0.1\",\"url\":\"/public/\",\"http_user_agent\":\"curl/7.74.0\",\"http_content_type\":\"text/html\",\"http_method\":\"GET\",\"protocol\":\"HTTP/1.1\",\"status\":200,\"length\":153}}"`
+
+HTTP fields in the `httpVals` are:
+
+|Http field                 | Description |
+|---------------------------|-------------|
+| hostname | IP address of the sender |
+| url | url requested on the server |
+| http_user_agent | Application used for HTTP |
+| http_content_type | type of content being returned by the server |
+| http_method | HTTP method used for the request |
+| protocol | HTTP protocol version used for the request or response |
+| status | HTTP status code|
+| length | size of the response body|
+
+As of now, the only supported layer 7 protocol is `HTTP`. More protocols will be supported in the near future.
```

Comment:
sorry, I didn't understand, will the http data be different in case of HTTP/2? I do not see the mentioning of HTTP/2 separately in Suricata document for HTTP event output.

Reply:
Given that the Suricata documentation has a different entry for HTTP/2 events, I thought it may be different: https://docs.suricata.io/en/latest/output/eve/eve-json-format.html#event-type-http2. Could you double-check?

Reply:
no we currently support HTTP 1.1, we can have HTTP2 as a future feature.